# Generated by Django 4.2.18 on 2025-02-13 06:28

from django.db import migrations


def _one_off_allele_liftover_error(apps, _schema_editor):
    """ I accidentally used the wrong key. This doesn't make much difference but things look slightly nicer if you
        use the right one """
    AlleleLiftover = apps.get_model("snpdb", "AlleleLiftover")

    records = []
    for allele_liftover in AlleleLiftover.objects.filter(error__message__isnull=True, error__error_message__isnull=False):
        allele_liftover.error["message"] = allele_liftover.error.pop("error_message")
        records.append(allele_liftover)

    if records:
        AlleleLiftover.objects.bulk_update(records, fields=["error"], batch_size=1000)


class Migration(migrations.Migration):

    dependencies = [
        ('snpdb', '0164_one_off_add_genome_build_slug'),
    ]

    operations = [
        migrations.RunPython(_one_off_allele_liftover_error)
    ]
